File: src/main/java/masonry/BrickListContainer.java
package masonry;

import masonry.api.Brick;
import masonry.api.BrickContainer;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class BrickListContainer<T extends Brick> implements BrickContainer<T> {

    private Size size;
    private List<T> items = new ArrayList<>();

    public BrickListContainer(int width, int height) {
        this.size = new Size(width, height);
    }

    @Override
    public Collection<T> getBricks() {
        return items;
    }

    public void addBrick(T brick) {
        this.items.add(brick);
    }

    public void removeBrick(T brick) {
        this.items.remove(brick);
    }

    @Override
    public int getHeight() {
        return size.height;
    }

    @Override
    public int getWidth() {
        return size.width;
    }
}

File: src/main/kotlin/net/scriptgate/masonry/BrickListContainer.kt
package net.scriptgate.masonry

import net.scriptgate.common.Point
import net.scriptgate.masonry.api.Brick
import net.scriptgate.masonry.api.BrickContainer
import java.util.*

class BrickListContainer<T : Brick>(width: Int, height: Int) : BrickContainer<T> {

    private val size: Point = Point(width, height)
    private val items = ArrayList<T>()

    override val bricks: Collection<T>
        get() = items

    override val height: Int
        get() = size.y

    override val width: Int
        get() = size.x

    fun addBrick(brick: T) {
        this.items.add(brick)
    }

    fun removeBrick(brick: T) {
        this.items.remove(brick)
    }
}
